resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java

```diff
@@ -681,26 +710,20 @@ private RequestConfig getCurrentConfiguration(final ClientInvocation request, fi
 
    public boolean isClosed()
    {
-      return closed;
+      return closed.get();
    }
 
-   public void close()
-   {
-      if (closed)
-         return;
+   @Override
+   public void close() {
+      cleanable.clean();
+   }
```

Comment:
I'm not clear about this. cleanable.clean() will set the AtomicBoolean closed to true. Then, if the CleanupAction gets called, closed.compareAndSet(false, true) returns false and client.close() doesn't happen. What am I missing?